File: content/problems/0016-3sum-closest/solutions/sort-two-pointers/Solution.java
class Solution {
    public int threeSumClosest(int[] nums, int target) {
        Arrays.sort(nums); int best = nums[0] + nums[1] + nums[2];
        for (int first = 0; first < nums.length - 2; first++) { int left = first + 1, right = nums.length - 1; while (left < right) { int sum = nums[first] + nums[left] + nums[right]; if (Math.abs((long)sum - target) < Math.abs((long)best - target)) best = sum; if (sum == target) return target; if (sum < target) left++; else right--; } }
        return best;
    }
}

File: content/problems/0016-3sum-closest/solutions/sort-two-pointers/Solution.kt
class Solution {
    fun threeSumClosest(nums: IntArray, target: Int): Int {
        nums.sort(); var best = nums[0] + nums[1] + nums[2]
        for (first in 0 until nums.size - 2) { var left = first + 1; var right = nums.lastIndex; while (left < right) { val sum = nums[first] + nums[left] + nums[right]; if (kotlin.math.abs(sum.toLong() - target) < kotlin.math.abs(best.toLong() - target)) best = sum; if (sum == target) return target; if (sum < target) left++ else right-- } }
        return best
    }
}
